testutils/src/main/java/com/amplifyframework/testutils/Await.java

```diff
@@ -134,28 +134,37 @@ private Await() {}
             @NonNull ResultErrorEmitter<R, E> resultErrorEmitter,
             @NonNull AtomicReference<R> resultContainer,
             @NonNull AtomicReference<E> errorContainer) {
-
-        CountDownLatch latch = new CountDownLatch(1);
-
-        resultErrorEmitter.emitTo(
-            result -> {
-                resultContainer.set(result);
-                latch.countDown();
-            }, error -> {
-                errorContainer.set(error);
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<Throwable> unexpectedErrorContainer = new AtomicReference<>();
+        final Thread thread = new Thread(() -> {
+            try {
+                resultErrorEmitter.emitTo(
+                    result -> {
+                        resultContainer.set(result);
+                        latch.countDown();
+                    }, error -> {
+                        errorContainer.set(error);
+                        latch.countDown();
+                    }
+                );
+            } catch (Throwable unexpectedFailure) {
+                unexpectedErrorContainer.set(unexpectedFailure);
                 latch.countDown();
             }
-        );
+        });
+        thread.setDaemon(true);
```

Comment:
is there any reason why you want these threads to persist? I feel like all threads should be stopped if the test suite that started it crashes due to a certain error that is unrelated to this thread.

Reply:
Ahhh!!! Good catch. I set out to make this say _`false`_, (`true` is the default anyway.) But, this does _not_ say `false`, does it.

Reply:
ah okay. that sounds like a good reason.